connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -0,0 +1,390 @@
+            throws InterruptedException, ExecutionException {
+        synchronized (this) {
+            return sourceAdminClient.describeAcls(ANY_TOPIC_ACL).values().get();
+        }
+    }
+
+    private Collection<TopicDescription> describeTopics(Collection<String> topics)
+            throws InterruptedException, ExecutionException {
+        synchronized (this) {
+            return sourceAdminClient.describeTopics(topics).all().get().values();
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    // use deprecated alterConfigs API for broker compatibility back to 0.11.0
+    private void updateTopicConfigs(Map<String, Config> topicConfigs)
+            throws InterruptedException, ExecutionException {
+        Map<ConfigResource, Config> configs = topicConfigs.entrySet().stream()
+            .filter(x -> shouldReplicateTopicConfigurationProperty(x.getKey()))
+            .collect(Collectors.toMap(x ->
+                new ConfigResource(ConfigResource.Type.TOPIC, x.getKey()), x -> x.getValue()));
+        log.trace("Syncing configs for {} topics.", configs.size());
+        synchronized (this) {
+            targetAdminClient.alterConfigs(configs).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e != null) {
+                    log.warn("Could not alter configuration of topic {}.", k.name(), e);
+                }
+            }));
+        }
+    }
+
+    private void updateTopicAcls(List<AclBinding> bindings)
+            throws InterruptedException, ExecutionException {
+        log.trace("Syncing {} topic ACL bindings.", bindings.size());
+        synchronized (this) {
+            targetAdminClient.createAcls(bindings).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e != null) {
+                    log.warn("Could not sync ACL of topic {}.", k.pattern().name(), e);
+                }
+            }));
+        }
+    }
+
+    private static Stream<TopicPartition> expandTopicDescription(TopicDescription description) {
+        String topic = description.name();
+        return description.partitions().stream()
+            .map(x -> new TopicPartition(topic, x.partition()));
+    }
+
+    private Map<String, Config> describeTopicConfigs(Set<String> topics)
+            throws InterruptedException, ExecutionException {
+        Set<ConfigResource> resources = topics.stream()
+            .map(x -> new ConfigResource(ConfigResource.Type.TOPIC, x))
+            .collect(Collectors.toSet());
+        synchronized (this) {
+            return sourceAdminClient.describeConfigs(resources).all().get().entrySet().stream()
+                .collect(Collectors.toMap(x -> x.getKey().name(), x -> x.getValue()));
+        }
+    }
+
+    Config targetConfig(Config sourceConfig) {
+        List<ConfigEntry> entries = sourceConfig.entries().stream()
+            .filter(x -> !x.isDefault() && !x.isReadOnly() && !x.isSensitive())
```

Comment:
Should we also filter configs with `STATIC_BROKER_CONFIG` as the source?

Reply:
I _think_ isDefault() will catch that case? Not sure.

Reply:
`isDefault()` only matches `ConfigSource.DEFAULT_CONFIG`. See https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java#L113-L118

Reply:
Ah, thanks @mimaison. Fixed.